sdk/keyvault/azsecrets/client.go

```diff
@@ -218,32 +216,36 @@ type DeleteSecretPoller struct {
 	client         *internal.KeyVaultClient
 	deleteResponse internal.KeyVaultClientDeleteSecretResponse
 	lastResponse   internal.KeyVaultClientGetDeletedSecretResponse
-	RawResponse    *http.Response
+	rawResponse    *http.Response
 }
 
 // Done returns true if the LRO has reached a terminal state
 func (s *DeleteSecretPoller) Done() bool {
-	return s.lastResponse.RawResponse != nil
+	if s.rawResponse == nil {
+		return false
+	}
+	return s.rawResponse.StatusCode == http.StatusOK
 }
 
 // Poll fetches the latest state of the LRO. It returns an HTTP response or error.(
 // If the LRO has completed successfully, the poller's state is updated and the HTTP response is returned.
 // If the LRO has completed with failure or was cancelled, the poller's state is updated and the error is returned.)
 func (s *DeleteSecretPoller) Poll(ctx context.Context) (*http.Response, error) {
+	var rawResp *http.Response
+	ctx = runtime.WithCaptureResponse(ctx, &rawResp)
 	resp, err := s.client.GetDeletedSecret(ctx, s.vaultUrl, s.secretName, nil)
+	s.rawResponse = rawResp
```

Comment:
I'm not sure what `GetDeletedSecret` does under the hood, but could there be a case where `err != nil` and `rawResp` is `nil`?  So you'd overwrite the last response with a `nil` one.  Maybe this is ok, just pointing it out.